cli/commands/stack/action.go

```diff
@@ -0,0 +1,242 @@
+package stack
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gruntwork-io/terragrunt/internal/experiment"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	getter "github.com/hashicorp/go-getter"
+
+	"github.com/gruntwork-io/terragrunt/internal/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+const (
+	generate         = "generate"
+	stackCacheDir    = ".terragrunt-stack"
+	defaultStackFile = "terragrunt.stack.hcl"
+	dirPerm          = 0755
+)
+
+// Run runs the stack command.
+func Run(ctx context.Context, opts *options.TerragruntOptions, subCommand string) error {
+	stacksEnabled := opts.Experiments[experiment.Stacks]
+	if !stacksEnabled.Enabled {
+		return errors.New("stacks experiment is not enabled use --experiment stacks to enable it")
+	}
+
+	if subCommand == "" {
+		return errors.New("No command specified")
```

Comment:
Sorry, I didn't think of this until now, but could you return a help message instead of only throwing an error? I think that's a better pattern to follow for commands with required arguments.